File: learn-spark/scala/wordcount/src/main/scala/wordcount.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object Wordcount {
    def main(args: Array[String]) {
        val inputFile = "file:///usr/local/spark-2.2.1/README.md"
        val conf = new SparkConf().setAppName("WordCount").setMaster("local[4]")
        val sc = new SparkContext(conf)
        val textFile = sc.textFile(inputFile)
        val wordcount = textFile.flatMap(line => line.split(" ")).map(word => (word, 1)).reduceByKey((a, b) => a + b)
        wordcount.foreach(println)

    }
}

File: learn-spark/python/base_demo/wordcount/wordcount.py
#!/usr/bin/python3
# -*- coding utf-8 -*-
from pyspark import SparkContext
sc = SparkContext('local', 'wordcount')
# hdfs dfs -put ./word.txt .
textFile = sc.textFile("word.txt")
wordcount = textFile.flatMap(lambda line: line.split(" "))\
    .map(lambda word: (word, 1))\
    .reduceByKey(lambda a, b: a + b)

wordcount.foreach(print)